libs/langchain/langchain/embeddings/azure_openai.py

```diff
@@ -38,7 +38,7 @@ class AzureOpenAIEmbeddings(OpenAIEmbeddings):
         For more: 
         https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id.
     """  # noqa: E501
-    azure_ad_token_provider: Union[str, None] = None
+    azure_ad_token_provider: Union[Callable[[], str], None] = None
```

Comment:
Is this only relevant for `AzureOpenAIEmbeddings`? Doesn't the same problem exist in `AzureOpenAI` completion and chat models?

Reply:
Yes, it has been partially fixed before, ref: https://github.com/langchain-ai/langchain/pull/14166